src/orchestrator.ts

```diff
@@ -205,26 +246,187 @@ export class Orchestrator implements Callbacks {
+            this.runningMessages[tick] = sourceRunner.part.onMessageProcessedCb(
+                streamIdentify.tick,
+                streamIdentify.channel,
+            )
+            const prom = new Promise(
+                (res) =>
+                    (this.connectingStreams[tick] = {
+                        promise: () => res(null),
+                        write: { write: async () => {}, close: async () => {} },
+                        log: this.logLevels[streamIdentify.channel],
+                    }),
+            )
+
+            targetInstantiator.streamMessage({
+                channel: streamIdentify.channel,
+                id: tick,
+                tick: tick,
+            })
+            await prom
+        } else {
+            this.logger.error(
+                `Receiving stream message for channel ${streamIdentify.channel} without a connected reader`,
+            )
+            const onEnd = sourceRunner.part.onMessageProcessedCb(
+                streamIdentify.tick,
+                streamIdentify.channel,
+            )
+
+            this.connectingStreams[tick] = {
+                promise: () => {},
+                write: {
+                    write: async () => {},
+                    close: async () => {
+                        onEnd()
+                    },
+                },
+            }
+        }
+
+        return tick
     }
 
     /**
-     * Handles streaming messages by forwarding them to all runners in the pipeline.
+     * Forwards streaming data chunks from source to target through the connecting stream.
      *
-     * @param {StreamMessage} msg - The stream message to process
+     * @param {number} tick - The message tick ID for the stream
+     * @param {AsyncIterable<StreamChunk>} stream - The stream of data chunks to forward
      * @returns {Promise<void>}
      */
-    async streamMessage(msg: StreamMessage) {
-        this.logger.debug('Got data stream message for channel ' + msg.channel)
-        await Promise.all(
-            this.pipeline.parts.map((part) =>
-                part.instantiator.streamMessage(msg),
-            ),
-        )
+    async forwardStream(tick: number, stream: AsyncIterable<StreamChunk>) {
+        const obj = this.connectingStreams[tick]
+
+        for await (const chunk of stream) {
+            if (obj.log) {
+                obj.log(decoder.decode(chunk.data!.data))
+            }
+            await obj.write.write(chunk.data!)
+        }
+
+        obj.write.close()
+    }
+
+    /**
+     * Establishes a connection for receiving streaming data.
+     * Links the stream writer to the connecting stream identified by the message tick.
+     *
+     * @param {number} id - The message tick ID for the stream connection
+     * @param {Sender<DataChunk>} write - The sender for writing data chunks to the stream
+     * @returns {Promise<void>}
+     */
+    async connectingReceivingStream(id: number, write: Sender<DataChunk>) {
```

Comment:
```suggestion
    async connectReceivingStream(id: number, write: Sender<DataChunk>) {
```

Reply:
This comment was resolved without any changes. Could you please explain your disagreement, @ajuvercr?

I’ll try to explain my reasoning: when writing a method name, I expect it to clearly state what the method does, in the imperative form, as if instructing the object or instance to perform a specific action. In this context, I assume the method’s purpose is to establish a connection for the receiving stream. In English, you would say, “connect with the receiving stream.” If we omit the unnecessary words (prepositions and articles), we are left with “connect receiving stream” or `connectReceivingStream`.